homeassistant/components/google_assistant/trait.py

```diff
@@ -634,20 +663,43 @@ def query_attributes(self):
                     "Lower bound for temperature range should be between "
                     "{} and {}".format(min_temp, max_temp))
 
+            attrs = self.state.attributes
+
+            data = {
+                ATTR_ENTITY_ID: self.state.entity_id,
+            }
+            # If the mode is auto==heatcool then google home will always
+            # send the range, even if range isn't supported by the device
+            if(climate.ATTR_TARGET_TEMP_HIGH in attrs and
+               climate.ATTR_TARGET_TEMP_LOW in attrs):
+                data[climate.ATTR_TARGET_TEMP_HIGH] = temp_high
+                data[climate.ATTR_TARGET_TEMP_LOW] = temp_low
+            else:
+                data[climate.ATTR_TEMPERATURE] = (temp_high + temp_low) / 2
+
             await self.hass.services.async_call(
-                climate.DOMAIN, climate.SERVICE_SET_TEMPERATURE, {
-                    ATTR_ENTITY_ID: self.state.entity_id,
-                    climate.ATTR_TARGET_TEMP_HIGH: temp_high,
-                    climate.ATTR_TARGET_TEMP_LOW: temp_low,
-                }, blocking=True)
+                climate.DOMAIN, climate.SERVICE_SET_TEMPERATURE, data,
+                blocking=True)
 
         elif command == COMMAND_THERMOSTAT_SET_MODE:
-            await self.hass.services.async_call(
-                climate.DOMAIN, climate.SERVICE_SET_OPERATION_MODE, {
-                    ATTR_ENTITY_ID: self.state.entity_id,
-                    climate.ATTR_OPERATION_MODE:
-                        self.google_to_hass[params['thermostatMode']],
-                }, blocking=True)
+            supported = self.state.attributes.get(ATTR_SUPPORTED_FEATURES)
+            mode = params['thermostatMode']
+
+            if (supported & climate.SUPPORT_ON_OFF != 0
```

Comment:
Here's there we handle "Hey google turn on/off \<device\>". It relies on SUPPORT_ON_OFF to activate, so will only happen for devices that have it.